homeassistant/components/google/config_flow.py

```diff
@@ -150,15 +153,16 @@ async def async_step_auth(
                 return self.async_abort(reason="oauth_error")
             self._device_flow = device_flow
 
+            exchange_finished_evt = asyncio.Event()
+            self._exchange_finished_task = self.hass.async_create_task(
+                exchange_finished_evt.wait()
+            )
+
             def _exchange_finished() -> None:
                 self.external_data = {
                     DEVICE_AUTH_CREDS: device_flow.creds
                 }  # is None on timeout/expiration
-                self.hass.async_create_task(
-                    self.hass.config_entries.flow.async_configure(
-                        flow_id=self.flow_id, user_input={}
-                    )
-                )
+                exchange_finished_evt.set()
 
             device_flow.async_set_listener(_exchange_finished)
```

Comment:
This can probably be simplified to use tasks in a follow up PR which I'll have a look at when you merge this.